lib/improver/utilities/cube_metadata.py

```diff
@@ -279,59 +287,196 @@ def update_attribute(cube, attribute_name, changes, warnings_on=False):
+
+    The cube will be modified in-place.
+
+    Args:
+        cube (iris.cube.Cube):
+            Cube containing cell methods that will be updated.
+        cell_method_definition (dict):
+            Dictionary potentially containing the keys: "comments", "coords",
+            "intervals" and "method", as supported by iris.coords.CellMethod.
+            The dictionary must specify an "action" keyword.
+
+    Raises:
+        ValueError: If no action is specified for the cell method, then raise
+                    an error.
+        ValueError: If no method is specified for the cell method, then raise
+                    an error.
+
+    """
+    if "action" not in cell_method_definition:
+        msg = ("No action has been specified within the cell method "
+               "definition. Please specify an action either 'add' or 'delete'."
+               "The cell method definition provided "
+               "was {}".format(cell_method_definition))
+        raise ValueError(msg)
+
+    if not cell_method_definition["method"]:
+        msg = ("No method has been specified within the cell method "
+               "definition. Please specify a method to describe "
+               "the name of the operation, see iris.coords.CellMethod."
+               "The cell method definition provided "
+               "was {}".format(cell_method_definition))
+        raise ValueError(msg)
+
+    for key in ["coords", "intervals", "comments"]:
+        if key not in cell_method_definition:
+            cell_method_definition[key] = ()
+
+    if not cell_method_definition["coords"]:
+        coords = ()
+    else:
+        coords = tuple([cell_method_definition["coords"]])
+
+    cell_method = iris.coords.CellMethod(
+        method=cell_method_definition["method"],
+        coords=coords,
+        intervals=cell_method_definition["intervals"],
+        comments=cell_method_definition["comments"])
+
+    cm_list = []
+    for cm in cube.cell_methods:
+        if cm == cell_method and cell_method_definition["action"] == "delete":
+            continue
+        cm_list.append(cm)
+
+    if cell_method_definition["action"] == "add":
+        if cell_method not in cube.cell_methods:
+            cm_list.append(cell_method)
+
+    cube.cell_methods = cm_list
+
+
 def amend_metadata(cube,
-                   new_diagnostic_name=None,
+                   name=None,
                    data_type=None,
-                   revised_coords=None,
-                   revised_attributes=None,
+                   coordinates=None,
+                   attributes=None,
+                   cell_methods=None,
+                   units=None,
                    warnings_on=False):
-    """Amend the metadata in the combined cube.
+    """Amend the metadata in the incoming cube. Please note that if keyword
+    arguments to this function are supplied by unpacking a dictionary, then
+    the keys of the dictionary need to correspond to the keyword arguments.
```

Comment:
I guess it's just because we don't use it in this way very often, so it's just a reminder to people. It could be removed if you felt strongly, but it doesn't hurt.